pilot/cmd/pilot-discovery/main.go

```diff
@@ -153,6 +153,8 @@ func init() {
 		"The domain serves to identify the system with spiffe")
 	discoveryCmd.PersistentFlags().StringVar(&serverArgs.Service.Consul.ServerURL, "consulserverURL", "",
 		"URL for the Consul server")
+	discoveryCmd.PersistentFlags().BoolVar(&serverArgs.Config.EnableConfigAnalysis, "enableAnalysis", serverArgs.Config.EnableConfigAnalysis,
+		"Enable config analysis service")
```

Comment:
Instead of a flag can you use the [feature package ]( pilot/pkg/features/pilot.go) for this purpose?

Reply:
I thought using pflags and viper was  the approved design (https://docs.google.com/document/d/1M-qqBMNbhbAxl3S_8qQfaeOLAiRqSBpSgfWebFBRuu8/edit#heading=h.qex63c29z2to).

Has this design been replaced?

Reply:
envvars are easier to patch with kustomize. It also fails more gracefully if we remove an option.

cc @costinm @howardjohn

Reply:
The design was never implemented, as far as I know. The current model we have been using is:
* Temporary flags (like this) are under feature package
* Permanent settings are in mesh config